src/libp2p/behaviour.rs

```diff
@@ -67,10 +68,23 @@ impl ForestBehaviour {
         network_name: &str,
     ) -> anyhow::Result<Self> {
         const MAX_ESTABLISHED_PER_PEER: u32 = 4;
-        const MAX_CONCURRENT_REQUEST_RESPONSE_STREAMS_PER_PEER: usize = 10;
+        static MAX_CONCURRENT_REQUEST_RESPONSE_STREAMS_PER_PEER: Lazy<usize> = Lazy::new(|| {
+            std::env::var("FOREST_MAX_CONCURRENT_REQUEST_RESPONSE_STREAMS_PER_PEER")
+                .ok()
+                .and_then(|it| {
+                    if let Ok(n) = it.parse() {
+                        if n > 0 {
+                            return Some(n);
+                        }
+                    }
+                    tracing::warn!("Failed to parse the `FOREST_MAX_CONCURRENT_REQUEST_RESPONSE_STREAMS_PER_PEER` environment variable value, a positive integer expected, got {it}.");
```

Comment:
I think we should just panic if it's not parsable. Warnings are easily overlooked, and lousy input should be punished.

Reply:
Fixed.

Reply:
Do we need the extra logic? Parsing a negative number to unsigned should fail. So we'd need something along the lines of `std::env::var("FOREST_MAX_CONCURRENT_REQUEST_RESPONSE_STREAMS_PER_PEER").ok().map(|v| v.parse().expect("silly user")).unwrap_or(10)`. Am I missing something?

Reply:
0 is invalid in this case

Reply:
Use a `NonZero`, then! :)

```rust
    std::env::var("FOREST_MAX_CONCURRENT_REQUEST_RESPONSE_STREAMS_PER_PEER")
        .ok()
        .map(|v| {
            v.parse::<NonZeroUsize>()
                .expect("Must be a positive number")
                .get()
        })
        .unwrap_or(10);

```

Reply:
Fixed.